docs/src/content/docs/guides/pages.mdx

```diff
@@ -0,0 +1,137 @@
+Use sub-folders to organize your files and to create multiple path segments.
+
+For example, the following file structure will generate pages at `example.com/hello-world` and `example.com/reference/faq`:
+
+import FileTree from '~/components/file-tree.astro';
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - hello-world.md
+      - reference/
+        - faq.md
+
+</FileTree>
+
+### Type-safe frontmatter
+
+All Starlight pages share a customizable [common set of frontmatter properties](/reference/frontmatter/) to control how the page appears:
+
+```md
+---
+title: Hello, World!
+description: This is a page in my Starlight-powered site
+---
+```
+
+If you forget anything important, Starlight will let you know.
+
+## Custom pages
+
+For advanced use cases, you can create custom pages in Astro’s `src/pages/` directory.
+This is helpful if you need to build pages with a completely custom layout or generate a page from an alternative data source.
+The `src/pages/` directory uses file-based routing and includes support for `.astro` files.
+
+Read more in the [“Pages” guide in the Astro docs](https://docs.astro.build/en/basics/astro-pages/).
+
+### Using Starlight’s design in custom pages
+
+To use the Starlight layout in custom pages, wrap your page content with the `<StarlightPage />` component.
+This can be helpful if you are generating content dynamically but still want to use Starlight’s design.
+
+```astro
+---
+// src/pages/custom-page/example.astro
+import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
+import CustomComponent from './CustomComponent.astro';
+---
+
+<StarlightPage frontmatter={{ title: 'My custom page' }}>
+	<p>This is a custom page with a custom component:</p>
+	<CustomComponent />
+</StarlightPage>
+```
+
+#### Props
+
+The `<StarlightPage />` component accepts the following props.
+
+##### `frontmatter` (required)
+
+**type:** `StarlightPageFrontmatter`
+
+Set the [frontmatter properties](/reference/frontmatter/) for this page, similar to frontmatter in Markdown pages.
+The [`title`](/reference/frontmatter/#title-required) property is required and all other properties are optional.
+
+The following properties differ from Markdown frontmatter:
+
+- The [`slug`](/reference/frontmatter/#slug) property is not supported and is automatically set based on the custom page’s URL.
+- The [`editUrl`](/reference/frontmatter/#editurl) option requires a URL to display an edit link.
+- The [`sidebar`](/reference/frontmatter/#sidebar) property is not supported. In Markdown frontmatter, this option allows customization of [autogenerated link groups](/reference/configuration/#sidebar), which is not applicable to pages using the `<StarlightPage />` component.
+
+##### `sidebar`
+
+**type:** `SidebarEntry[] | undefined`  
+**default:** the sidebar generated based on the [global `sidebar` config](/reference/configuration/#sidebar)
+
+Provide a custom site navigation sidebar for this page.
```

Comment:
Just pointing out that the structure of all of these definitions is an action the user might want to take, vs e.g.

"A property indicating whether Starlight's default global sidebar is used" or "The sidebar entry to use"

Also, I have no idea how I would set this. I think an example here would be helpful to show what that looks like.

Reply:
> the structure of all of these definitions is an action the user might want to take

That was intentional. Is it a good or bad thing? 😅 

+1 on an example being helpful.

Reply:
Ah, so the example prompt was very helpful! Turns out using this is not very fun, which I hadn’t tested 😅 

Basically the types are quite strict and there’s none of the helpful massaging that you get if you’re setting a sidebar in the Starlight config.

An example:

```astro
<StarlightPage
	frontmatter={{ title: 'Demo Page' }}
	sidebar={[
		{
			type: 'link',
			href: '/',
			label: 'Home',
			isCurrent: false,
			badge: undefined,
			attrs: {},
		},
		{
			type: 'group',
			label: 'Pages',
			entries: [
				{
					type: 'link',
					href: '/demo/',
					label: 'Demo',
					isCurrent: true,
					badge: undefined,
					attrs: {},
				},
			],
			badge: undefined,
			collapsed: false,
		},
	]}
/>
```

There’s a lot of boilerplate here, it could probably be more like this with some nice defaults:

```astro
<StarlightPage
	frontmatter={{ title: 'Demo Page' }}
	sidebar={[
		{ href: '/', label: 'Home' },
		{
			label: 'Pages',
			entries: [
				{ href: '/demo/', label: 'Demo', isCurrent: true },
			],
		},
	]}
/>
```

Wondering if we could just hold off documenting `sidebar` and ship a follow-up release that adds a nicer API and we document it then

Reply:
Pulled the trigger on that — will follow up in a future PR.

Reply:
Random thoughts: If we don't like the API, wouldn't it be better to do the change now rather than later and force all plugins and `<StarlightPage/>` users to manually do the change? This will definitely be used if released as-is or people will ask for it considering all the broken plugins or previous `<Page>` users like Biome were specifying custom sidebars and it's been broken since October. I'm not sure if it's a good idea to release it as-is after all this time and then change it to an entirely different API in a few PRs.